docs/src/modules/components/MarkdownDocs.js

```diff
@@ -0,0 +1,296 @@
+        )}
+        {rendered.map((renderedMarkdownOrDemo, index) => {
+          if (typeof renderedMarkdownOrDemo === 'string') {
+            return (
+              <React.Fragment key={index}>
+                <MarkdownElement renderedMarkdown={renderedMarkdownOrDemo} />
+              </React.Fragment>
+            );
+          }
+
+          if (renderedMarkdownOrDemo.component) {
+            const name = renderedMarkdownOrDemo.component;
+            const Component = srcComponents?.[name];
+
+            if (Component === undefined) {
+              throw new Error(`No component found at the path ${path.join('docs/src', name)}`);
+            }
+
+            return (
+              <React.Fragment key={index}>
+                <Component {...renderedMarkdownOrDemo} markdown={localizedDoc} />
+              </React.Fragment>
+            );
+          }
+
+          const name = renderedMarkdownOrDemo.demo;
+          const demo = demos?.[name];
+          if (demo === undefined) {
+            const errorMessage = [
+              `Missing demo: ${name}. You can use one of the following:`,
+              Object.keys(demos),
+            ].join('\n');
+
+            if (userLanguage === 'en') {
+              throw new Error(errorMessage);
+            }
+
+            if (process.env.NODE_ENV !== 'production') {
+              console.error(errorMessage);
+            }
+
+            const warnIcon = (
+              <span role="img" aria-label={t('emojiWarning')}>
+                ⚠️
+              </span>
+            );
+            return (
+              <div key={index}>
+                {/* eslint-disable-next-line material-ui/no-hardcoded-labels */}
+                {warnIcon} Missing demo `{name}` {warnIcon}
+              </div>
+            );
+          }
+
+          const splitLocationBySlash = location.split('/');
+          splitLocationBySlash.pop();
+          const fileNameWithLocation = `${splitLocationBySlash.join('/')}/${name}`;
+
+          return (
+            <Demo
+              key={index}
+              mode={theme.palette.mode}
+              demo={{
+                raw: demo.raw,
+                js: demoComponents[demo.module] ?? noComponent(demo.module),
+                scope: demos.scope,
+                jsxPreview: demo.jsxPreview,
+                rawTS: demo.rawTS,
+                tsx: demoComponents[demo.moduleTS] ?? null,
+                gaLabel: fileNameWithLocation.replace(/^\/docs\/data\//, ''),
+              }}
+              disableAd={disableAd}
+              demoOptions={renderedMarkdownOrDemo}
+              githubLocation={`${process.env.SOURCE_CODE_REPO}/blob/v${process.env.LIB_VERSION}${fileNameWithLocation}`}
+            />
+          );
+        })}
+        <ImageViewer
+          open={imageViewerOpen}
+          onClose={handleImageViewerClose}
+          maxWidth="lg"
+          fullWidth
+        >
+          <DialogContent className={classes.fullSizeImageDialog}>
```

Comment:
I think that we could customize this:

Current:

<img width="881" alt="Screenshot 2023-05-20 at 00 15 47" src="https://github.com/mui/mui-toolpad/assets/3165635/757b2af2-072b-4a0a-a4c6-4c3d5caa4e56">

What it could look like:

<img width="874" alt="Screenshot 2023-05-20 at 00 18 31" src="https://github.com/mui/mui-toolpad/assets/3165635/5b137d0e-8ba8-4788-be3e-dbbdb84d3b98">